clients/src/main/java/org/apache/kafka/common/protocol/types/SchemaVisitor.java

```diff
@@ -14,14 +14,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.kafka.common.protocol;
+package org.apache.kafka.common.protocol.types;
 
-import org.apache.kafka.common.protocol.types.ArrayOf;
-import org.apache.kafka.common.protocol.types.Schema;
-import org.apache.kafka.common.protocol.types.Type;
-
-public interface SchemaVisitor {
-    void visit(Schema schema);
-    void visit(ArrayOf array);
-    void visit(Type field);
+/**
+ * Override one or more of the visit methods with the desired logic.
+ */
+public abstract class SchemaVisitor {
```

Comment:
I guess we could also move this into `Schema` and just call it `Visitor`?

Reply:
Yeah, sounds good.